scripts/functional-tests.sh

```diff
@@ -124,7 +135,7 @@ cleanup() {
 }
 
 init_test_script
-trap 'cleanup "${?}"; rm -rf "${TEST_DIR}"' EXIT QUIT INT TERM
+trap 'cleanup "${?}"; test -z $SKIP_TEST_DIR_CLEANUP && rm -rf "${TEST_DIR}"' EXIT QUIT INT TERM
```

Comment:
Still not sure we have to have a separate variable for `SKIP_TEST_DIR_CLEANUP`, or we could just detect that `TEST_DIR` was passed by the caller and avoid cleaning up the directory then.

Reply:
Ok you sold me on this.